common/main/java/com/couchbase/lite/internal/core/C4Log.java

```diff
@@ -86,10 +142,82 @@ public final void forceCallbackLevel(@NonNull LogLevel logLevel) {
         setCoreCallbackLevel();
     }
 
+    @VisibleForTesting
+    // This, apparently, should be the inverse of LOGGING_DOMAINS_FROM_C4
+    public final void setC4LogLevel(@NonNull EnumSet<LogDomain> domains, @NonNull LogLevel level) {
+        final int c4Level = getC4LevelForLogLevel(level);
+        for (LogDomain domain: domains) {
+            switch (domain) {
+                case DATABASE:
+                    setLevels(
+                        c4Level,
+                        C4Constants.LogDomain.DATABASE,
+                        C4Constants.LogDomain.SQL,
+                        C4Constants.LogDomain.ZIP);
+                    break;
+
+                case LISTENER:
+                    setLevels(c4Level, C4Constants.LogDomain.LISTENER);
+                    break;
+
+                case NETWORK:
+                    setLevels(
+                        c4Level,
+                        C4Constants.LogDomain.BLIP,
+                        C4Constants.LogDomain.BLIP_MESSAGES,
+                        C4Constants.LogDomain.TLS,
+                        C4Constants.LogDomain.WEB_SOCKET);
+                    break;
+
+                case QUERY:
+                    setLevels(c4Level, C4Constants.LogDomain.QUERY);
+                    break;
+
+                case REPLICATOR:
+                    setLevels(
+                        c4Level,
+                        C4Constants.LogDomain.CHANGES,
+                        C4Constants.LogDomain.SYNC,
+                        C4Constants.LogDomain.SYNC_BUSY);
+                    break;
+
+                default:
+                    logInternal(
+                        getC4DomainForLoggingDomain(LogDomain.DATABASE),
+                        c4Level,
+                        "Unexpected log domain: " + domain);
+                    break;
+            }
+        }
+    }
+
```

Comment:
The rest of these, too.  I'll bet they don't need to be public anymore...

Reply:
one/two of these methods are used by LogTest also